docs/ResourceInterface.md

```diff
@@ -24,121 +26,158 @@ public function getId(): int
-
-The `transformer` method must return new transformer object that must inherit from [Sowl\JsonApi\AbstractTransformer](../src/AbstractTransformer.php).
-
-
+### transformer()
+Returns a new transformer instance.
 ```php
 public static function transformer(): AbstractTransformer
 {
     return new UserTransformer();
 }
 ```
 
-Example of simple transformer implementation:
+This Package uses [Fractal](https://fractal.thephpleague.com/) for serialization
+of entities/resources as JSON:API responses.
+
+[Fractal Transformers Documentation](https://fractal.thephpleague.com/transformers/)
+
+Transformer classes should extend the
+[`Sowl\JsonApi\AbstractTransformer`](/src/AbstractTransformer.php) class.
+
+[User Transformer Example](./examples/UserTransformer.php)
+
+### relationships()
+Returns the resource relationships.
+
 ```php
-class UserTransformer extends AbstractTransformer
+public static function relationships(): RelationshipsCollection
 {
-    public function transform(User $user): array
-    {
-        return [
-            'id' => $user->getId(),
-            'name' => $user->getName(),
-            'email' => $user->getEmail(),
-        ];
-    }
+    return new RelationshipsCollection();
 }
 ```
 
-## relationships()
-Method must return the resource relationships definitions.
-We need this method so that we can automatically serve all the relationships endpoints.
+This method must return an instance of
+[`Sowl\JsonApi\Relationships\RelationshipsCollection`](src/Relationships/RelationshipsCollection.php).
 
-Method must return [Sowl\JsonApi\Relationships\RelationshipsCollection](../src/Relationships/RelationshipsCollection.php).
+Two types of relationships exist, `To-One` and `To-Many`.
+
+#### To-One Relationship
+Returns a single resource in the responses.
 
-Example of resource without relationships:
 ```php
+// User Entity
 public static function relationships(): RelationshipsCollection
 {
-    return new RelationshipsCollection();
+    return new RelationshipsCollection([
+        ToOneRelationship::create('country', Country::class)
+    ]);
 }
 ```
 
-There are 2 types of relationships `To-One` and `To-Many` relationships.
+The `ToOneRelationship::create()` method accepts 3 parameters:
+
+- `name` (required):
+
+Name of the relationship that will be used in the endpoint:
+
+i.e: `GET /user/1/{name}` or `GET /user/1/relationships/{name}`
+
+i.e: `GET /user/1/country` or `GET /user/1/relationships/country`
+
+- `class` (required):
 
-### To-One
-To define new "To-One" relationship we must add new [Sowl\JsonApi\Relationships\ToOneRelationship](../src/Relationships/ToOneRelationship.php).
+The entity (resource) class being the relation.
 
-To-One relationship returns single resource in the responses.
+- `property` (optional):
 
-2 required parameters must be provider and 1 optional.
+Allow to set a custom value for the `{name}` for custom endpoints.
```

Comment:
It's not working like that.

`{name}` - is the endpoint name and a name in the `data.relationships.{name} in the responses.
`{property}` - Default: {name} is the Entity property name that will be updated. 

The property is not for creating custom endpoints, it's for allowing custom {name} if the {name} not matching the entity property.